engine/runtime/src/main/java/org/enso/interpreter/runtime/TruffleCompilerContext.java

```diff
@@ -40,14 +40,14 @@
 final class TruffleCompilerContext implements CompilerContext {
 
   private final EnsoContext context;
-  private final TruffleLogger compiler;
-  private final TruffleLogger serializationManager;
+  private final TruffleLogger compilerLogger;
+  private final TruffleLogger serializationManagerLogger;
   private final RuntimeStubsGenerator stubsGenerator;
 
   TruffleCompilerContext(EnsoContext context) {
     this.context = context;
-    this.compiler = context.getLogger(Compiler.class);
-    this.serializationManager = context.getLogger(SerializationManager.class);
+    this.compilerLogger = context.getLogger(Compiler.class);
```

Comment:
I left this logger as-is as I haven't seen the same illegal state failures as with SerializationManager, potentially because (de-)serializationis is typically run in a separate thread. We could change it as well, I'm open for suggestions.

Reply:
I think we want to bound all loggers to engine. No need to have two `getLogger` and `getLoggerBoundToEngine` method.